cordova-js-src/platform.js

```diff
@@ -56,9 +57,9 @@ module.exports = {
             // OR cordova.require('cordova/platform').activationContext
             // activationContext:{type: actType, args: args};
             var activationHandler = function (e) {
-                var args = e.detail.arguments;
-                var actType = e.detail.type;
-                platform.activationContext = { type: actType, args: args };
+                // Making all the details available as activationContext
+                platform.activationContext = utils.clone(e.detail); 
+                platform.activationContext.args = e.detail.arguments;       /* for backwards compatibility */
```

Comment:
That's true. Users might need to modify the object so I guess I will pass the actual object instead of cloning it.